pkg/strace/tracer.go

```diff
@@ -164,6 +164,11 @@ var traceActive uint32
 // recordCallback is called every time a process event happens with the process
 // in a stopped state.
 func Trace(c *exec.Cmd, recordCallback ...EventCallback) error {
+	return TraceWithSecComp(c, false, recordCallback...)
+}
+
+// TraceWithSecComp traces `c` and any children c clones with the option to enable seccomp.
+func TraceWithSecComp(c *exec.Cmd, secComp bool, recordCallback ...EventCallback) error {
```

Comment:
This naming confuses me a bit. If I see the name TraceWithSeccomp, I would assume it is tracing with SecComp. 

I wonder if we should not just call this New(), following common Go naming? 

The other problem is that we're baking in more and more linux dependencies, and this in principle could run on freebsd etc., but that's a problem for another day I guess. Technically, however, the ptrace-based on could be portable across unix, but a seccomp based one is harder.

Reply:
Using `New` might be a better choice.

Since the seccomp filter is implemented at the user code level, it should help alleviate dependency concerns for now (unless I’m mistaken).

Reply:
Changed it to `New()`

Reply:
Technically it can be done with Option pattern, like WithSeccomp() and so on, but I agree it a problem for another day.